alicloud/resource_alicloud_ots_table.go

```diff
@@ -0,0 +1,245 @@
+	if primaryKey4Name != nil {
+		tableMeta.AddPrimaryKeyColumn(primaryKey4Name.(string), primaryKey4Type)
+	}
+
+	tableOption := new(tablestore.TableOption)
+	tableOption.TimeToAlive = d.Get("time_to_live").(int)
+	tableOption.MaxVersion = d.Get("max_version").(int)
+	reservedThroughput := new(tablestore.ReservedThroughput)
+	//reservedThroughput.Readcap = d.Get("read_cap").(int)
+	//reservedThroughput.Writecap = d.Get("write_cap").(int)
+
+	createTableRequest := new(tablestore.CreateTableRequest)
+	createTableRequest.TableMeta = tableMeta
+	createTableRequest.TableOption = tableOption
+	createTableRequest.ReservedThroughput = reservedThroughput
+	_, err := client.CreateTable(createTableRequest)
+	if err != nil {
+		return fmt.Errorf("Failed to create table with error: %s", err)
+	} else {
+		log.Printf("Create table finished")
+	}
+
+	// Need to set id before calling read method or terraform.state won't be generated.
+	d.SetId(table_name)
+	//return resourceAliyunOtsTableUpdate(d, meta)
+	return resourceAliyunOtsTableRead(d, meta)
+}
+
+func resourceAliyunOtsTableRead(d *schema.ResourceData, meta interface{}) error {
+	log.Println("Describing OTS Table...")
+	client := getOtsClient()
+
+	describeTableReq := new(tablestore.DescribeTableRequest)
+	describeTableReq.TableName = d.Get("table_name").(string)
+
+	describ, err := client.DescribeTable(describeTableReq)
+
+	if err != nil {
+		fmt.Println("failed to update table with error: %s", err)
+	} else {
+		d.Set("table_name", describ.TableMeta.TableName)
+		d.Set("time_to_live", describ.TableOption.TimeToAlive)
+		d.Set("max_version", describ.TableOption.MaxVersion)
+		d.Set("read_cap", describ.ReservedThroughput.Readcap)
+		d.Set("write_cap", describ.ReservedThroughput.Writecap)
+		log.Println("DescribeTable finished. Table meta: %d, %d", describ.TableOption.MaxVersion, describ.TableOption.TimeToAlive)
+	}
+
+	return nil
+}
+
+func resourceAliyunOtsTableUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Println("Updating OTS Table...")
+	client := getOtsClient()
+
+	update := false
+
+	updateTableReq := new(tablestore.UpdateTableRequest)
+	table_name := d.Get("table_name").(string)
+	updateTableReq.TableName = table_name
+	updateTableReq.TableOption = new(tablestore.TableOption)
+
+	if d.HasChange("time_to_live") {
+		update = true
+		time_to_live := d.Get("time_to_live").(int)
+		log.Printf("time_to_live changed to: %d", time_to_live)
+		updateTableReq.TableOption.TimeToAlive = time_to_live
+	}
+
+	if d.HasChange("max_version"){
+		update = true
+		max_version := d.Get("max_version").(int)
+		log.Printf("time_to_live changed to: %d", max_version)
+		updateTableReq.TableOption.MaxVersion = max_version
+	}
+
+	if update {
+		_, err := client.UpdateTable(updateTableReq)
+
+		if err != nil {
+			log.Println("failed to update table with error: %s", err)
+		} else {
+			log.Println("update finished")
+			d.SetId(table_name)
```

Comment:
There is no need to set id in update method.

Reply:
Removed.